### PR TITLE
Show pseudonymized names

### PR DESCRIPTION
**Pull request by Mikael-Lenander**
This turned out to be very difficult. The only way I came up with is to modify the user instance's fields without storing the changes to the database. I don't know if this is the cleanest way to do it but the upside is that is doesn't require changing the API of user of userprofile. I couldn't figure out a way to pseudonymize the 'All results' page since it doesn't fetch the user data in the view but through a JavaScript script so I can't access the request.session object which contains a boolean value 'pseudonymize' that determines whether personal data should be pseudonymized. Also, the forms are difficult to pseudonymize without changing the form field values which sounds dangerous.

#533

### CODE REVIEW

File: exercise/staff_views.py [hunk 1]
```diff
@@ -66,6 +75,9 @@ def get_common_objects(self) -> None:
                 Prefetch('submitters', UserProfile.objects.prefetch_tags(self.instance)),
             )
         )
+        for submission in qs:
+            format_submission(submission, self.pseudonymize)
+            print(submission.submitters)
```
Comment (murhum1) [inline exercise/staff_views.py:80]: Should remove debug print

**Review by murhum1 — CHANGES_REQUESTED** [review]
Just wrote up comments earlier and didn't actually submit them, sorry! New to this review business. Anyway, comments up now.

File: userprofile/pseudonymize.py
[the remark this answers is not in the document]
Comment (mikaelGusse) [reply userprofile/pseudonymize.py:1]: Added a seed such that all the fields that are generated are consistent per user now

File: external_services/templatetags/external_services.py [hunk 2]
```diff
@@ -1,4 +1,5 @@
 import string
+
```
[the remark this answers is not in the document]
Comment (mikaelGusse) [reply external_services/templatetags/external_services.py]: Fixed

File: exercise/templates/exercise/staff/_assessment_panel.html [hunk 3]
```diff
@@ -10,9 +10,15 @@
...
 				<span class="glyphicon glyphicon-ok" aria-hidden="true"></span>
-				{% blocktranslate trimmed with user=submission.grader.user.get_full_name time=submission.grading_time %}
-				ASSESSED_BY -- {{ user }}, {{ time }}
-				{% endblocktranslate %}
+				{% if not pseudonymize %}
+					{% blocktranslate trimmed with user=submission.grader.user.get_full_name time=submission.grading_time %}
+					ASSESSED_BY -- {{ user }}, {{ time }}
+					{% endblocktranslate %}
+				{% else %}
+					{% blocktranslate trimmed with user="Somebody" time=submission.grading_time %}
```
[the remark this answers is not in the document]
Comment (mikaelGusse) [reply exercise/templates/exercise/staff/_assessment_panel.html]: Done

File: exercise/templates/exercise/_submission_info.html [hunk 4]
```diff
@@ -33,14 +33,18 @@ <h4 class="panel-title">{% translate "SUBMISSION_INFO" %}</h4>
...
+					  <script src="{% static 'django_colortag.js' %}"></script>
+					  <script src="{% static 'add_tagging_dropdown.js' %}"></script>
+					  {% endif %}
+					  {% profiles submission.submitters.all instance is_teacher %}
+					  {% else %}
+					  {{ submission.submitters.all|names }}
+				  {% endif %}
 			  {% else %}
-			  {{ submission.submitters.all|names }}
+				Somebody
```
[the remark this answers is not in the document]
Comment (mikaelGusse) [reply exercise/templates/exercise/_submission_info.html]: Done

**Comment by mikaelGusse** [comment]
> Just wrote up comments earlier and didn't actually submit them, sorry! New to this review business. Anyway, comments up now.

Thanks for the comments! Just went through them, do you think the seed system there is ok for keeping the pseudonymizations consistent?

**Review by murhum1 — APPROVED** [review]
Thanks! Looks fine now.

**Pull request merged**